libs/screen/image.cpp

```diff
@@ -412,6 +412,15 @@ void _mapRect(Image_ img, int xy, int wh, Buffer c) {
     mapRect(img, XX(xy), YY(xy), XX(wh), YY(wh), c);
 }
 
+//%
+bool _equals(Image_ img, Image_ other) {
```

Comment:
just call it `equals()`, put it in ImageMethods namespace, and skip the helper - it will be added to the interface. The reason _mapRect() is done with helper, is because it has more than 4 args

Reply:
I might have got in my own head a bit on this, I was doing that to have a null / undefined check for the ``other`` param in the ts.  Is there a macro somewhere to check that in cpp, I'm not seeing how to compare to / check against the tagged values right away.

Reply:
Currently, if you pass null/undefined to a C++ method, it will 980. I'm thinking we should have some annotation to allow null to be passed to C++.

Reply:
Oh, this only applies for custom reference types. If C++ takes integer it will get `0` for `null`, if it takes a string it will get `"null"`

Reply:
Ah. So is there a better way to structure this then for now, if I want to just return false on ``validImage.equals(undefined)`` instead of 980? That's what I was trying to oavoid with the helper / ts split, definitely feels a bit wonky / a bit fragile as is though. I guess the annotation would just convert ``null`` / ``undefined`` for custom types to ``nullptr`` so it can be checked in the function itself?

Reply:
Correct, and correct.

Reply:
let me see if I can update pxt to support null today